### DISCUSSION TITLE
Tiempo ejecución mac edd-docker

### DISCUSSION THREAD

**Discussion by nonoam** [IIC2133-PUC/2024-2#53] [Tarea 0] [answered]
Hola, hay varias preguntas sobre el tiempo de ejecución, pero no encontré una que se respondiera mi caso. 

Ocurre que cuando ejecuto mi programa con el último test para ver el tiempo de ejecución dentro del contenedor de docker (estoy en MAC), se demora cerca de 16seg en su ejecución. En cambio si lo ejecuto en la terminal nativa (fuera del contenedor docker) se demora apenas 4.1seg. 

Aquí dejo unas fotos de lo mencionado: 

En container docker:
<img width="153" alt="Screenshot 2024-08-22 at 12 24 17 PM" src="https://github.com/user-attachments/assets/7667d8c9-1b81-49c1-b851-01207d0735ae">

Terminal nativa:
<img width="1048" alt="Screenshot 2024-08-22 at 12 24 38 PM" src="https://github.com/user-attachments/assets/9705efa9-4a63-4233-8b24-0dd7d7142cad">

Mi pregunta es: ¿Esto es normal o hay algo que estoy haciendo mal (a parte de la optimización de mi código)?

De todas maneras tengo presente que debo optimizar mi código, para que el tiempo de búsqueda sea considerablemente menor en ambos escenarios explicados anteriormente.

**Comment by sebacarrascop** [comment, marked as answer]
hola!, intente correr tu código en el servidor, pero al menos lo que tienes pusheado me da Segmentation fault, por si puedes pushear tus avances para revisarlo de primera mano.

Por otro lado, es normal que correr el código dentro de docker sea algo más lento que de forma local, ya que docker limita los recursos para la ejecución. Si bien el servidor suele ser mucho más rápido que la ejecución con docker, la diferencia que tienes actualmente es alta, por lo que deben haber problemas respecto a las complejidades que usas en tus algoritmos para que provoquen estas diferencias.

Cuando pushees tus avances podría ver el problema con mayor detalle.
Saludos!

**Reply by nonoam** [reply]
Holaa, disculpa la demora, pero ya pushee mis avances. Por si me podría confirmar si solamente es algo de mi computadora dado los recursos o meramente mi código :D

**Reply by sebacarrascop** [reply]
hola!, revisa #59
